src/Keter/App.hs

```diff
@@ -133,36 +141,38 @@ data AppStartConfig = AppStartConfig
     , ascHostManager    :: !HostManager
     , ascPortPool       :: !PortPool
     , ascPlugins        :: !Plugins
-    , ascLog            :: !(LogMessage -> IO ())
     , ascKeterConfig    :: !KeterConfig
     }
 
-withConfig :: AppStartConfig
-           -> AppId
+withConfig :: AppId
            -> AppInput
-           -> (Maybe FilePath -> BundleConfig -> Maybe EpochTime -> IO a)
-           -> IO a
-withConfig _asc _aid (AIData bconfig) f = f Nothing bconfig Nothing
-withConfig asc aid (AIBundle fp modtime) f = bracketOnError
-    (unpackBundle asc fp aid)
-    (\(newdir, _) -> removeDirectoryRecursive newdir)
-    $ \(newdir, bconfig) -> f (Just newdir) bconfig (Just modtime)
-
-withReservations :: AppStartConfig
-                 -> AppId
+           -> (Maybe FilePath -> BundleConfig -> Maybe EpochTime -> KeterM AppStartConfig a)
+           -> KeterM AppStartConfig a
+withConfig _aid (AIData bconfig) f = f Nothing bconfig Nothing
+withConfig aid (AIBundle fp modtime) f = do
+    withRunInIO $ \rio ->
+        bracketOnError
+            (rio $ unpackBundle fp aid)
+            (\(newdir, _) -> removeDirectoryRecursive newdir)
+            $ \(newdir, bconfig) -> rio $ f (Just newdir) bconfig (Just modtime)
```

Comment:
I'm ont sure all these changes are the result of an auto formatter, but it makes reviewiing harder if that's the case, I recomend disabling it.